crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1807,6 +1807,8 @@ impl<'db> KnownClass {
 pub enum KnownInstanceType<'db> {
     /// The symbol `typing.Literal` (which can also be found as `typing_extensions.Literal`)
     Literal,
+    /// The symbol `typing.Optional` (which can also be found as `typing_extensions.Literal`)
```

Comment:
Should this comment be repeated on every special form? Most of them have this properties.

Reply:
I find the doc-comments helpful. It's true that many will likely be quite similar, but the comment for an enum variant shows up in the tooltip when you hover over that enum variant in your IDE with the ruff repo open